### DISCUSSION TITLE
i386?

### DISCUSSION THREAD

**Discussion by AndreRH** [jart/blink#82] [Q&A]
Hi,

The readme indicates a bit that i386 emulation is also possible with blink, if that's correct my questions are:

- does the x86_64 "limitation" only apply to the loader?
- is i386 emulation on par with x86_64 emulation?

**Comment by jart** [comment]
The Blink codebase supports the i386 ISA, but Blink doesn't implement the Linux i386 `int $0x80` system call abi for example. Since there's no system call interface in Blink for i386, the `blink/loader.c` code will reject an i386 ELF binary if you try to run one. Where you *can* currently use i386 code would be if you used `blinkenlights -r foo.bin` to load a real mode program from BIOS, and then the program put itself in protected mode. In that case Blink would be doing i386.

**Reply by AndreRH** [reply]
Nice! The emulator part is what I need. I'll give it a try and report back
